lib/predicates/unique.rb

```diff
@@ -60,10 +68,10 @@ def comparison_for(field, value, klass)
       end
     else
       # non-text comparison
-      klass.send(:sanitize_sql, {field => value})
+      klass.send(:sanitize_sql, ["#{quoted_field} = ?", value])
```

Comment:
silences this deprecation message in 4.2:

`DEPRECATION WARNING: sanitize_sql_hash_for_conditions is deprecated, and will be removed in Rails 5.0`